browser-extension/src/ui/index.tsx

```diff
@@ -0,0 +1,361 @@
+        settings: this.state.settings,
+      },
+      async () => {
+        if (this.props.activeTab) {
+          const tabId = Number(this.props.activeTab.id);
+          if (chrome.scripting) {
+            chrome.scripting.executeScript({
+              target: {
+                tabId,
+              },
+              function: () => {
+                window.location.reload();
+              },
+            });
+          } else {
+            chrome.tabs.executeScript(tabId, {
+              code: 'window.location.reload();',
+            });
+          }
+        }
+      }
+    );
+  }
+
+  render() {
+    const { urlFilter, exporters } = this.state.settings;
+
+    const classes = this.props.classes;
+
+    const saveLabel =
+      this.props.app === AppType.POPUP ? Label.SAVE_AND_RELOAD : Label.SAVE;
+
+    return (
+      <React.Fragment>
+        <CssBaseline />
+        <AppBar position="absolute" color="default" className={classes.appBar}>
+          <Toolbar>
+            {this.props.app === AppType.OPTIONS ? (
+              <Typography variant="h6" color="inherit" noWrap>
+                {capitalCase(packageJson.name)} ({packageJson.version})
+              </Typography>
+            ) : (
+              <React.Fragment>
+                <Button
+                  variant="contained"
+                  color="primary"
+                  size="small"
+                  onClick={this.handleSaveSettings}
+                >
+                  {saveLabel}
+                </Button>
+                <Typography className={classes.title} />
+                <Link href="#" onClick={this.openOptionsPage}>
+                  <LaunchIcon></LaunchIcon>
+                </Link>
+              </React.Fragment>
+            )}
+          </Toolbar>
+        </AppBar>
+        <main className={classes.layout}>
+          <Paper className={classes.paper}>
+            <Typography
+              component="h1"
+              variant="h6"
+              color="primary"
+              gutterBottom
+            >
+              Injection Settings
+            </Typography>
+            <Grid container spacing={3}>
+              <Grid item xs={12}>
+                <TextField
+                  id="urlFilter"
+                  label="URL Filter"
+                  fullWidth
+                  variant="outlined"
+                  margin="dense"
+                  value={urlFilter}
+                  onChange={this.handleFilterChange}
+                  helperText='Injection is only applied if the URL contains the given filter. Use "*" to match every URL.'
+                />
+              </Grid>
+            </Grid>
+          </Paper>
+          <Paper className={classes.paper}>
+            <Typography
+              component="h1"
+              variant="h6"
+              color="primary"
+              gutterBottom
+            >
+              Exporter Settings
+            </Typography>
+            <Grid container spacing={2}>
```

Comment:
I would split this into smaller components 
So it could look like this
```
<ExporterOption label="Console" url="https://www.npmjs.com/package/@opentelemetry/tracing" toggle={exporters[ExporterType.CONSOLE]}/>

<ExporterOption label="Zipkin" url="https://www.npmjs.com/package/@opentelemetry/tracing" toggle={exporters[ExporterType.ZIPKIN]}/>

<ButtonSave onClick={this.handleSaveSettings}/>
```
etc.
It will be more readable